README.md

```diff
@@ -144,33 +144,105 @@ $ make -C build.assets build-binaries
 
 ### Local Build
 
-To perform a build on your host, ensure you have installed Go. In order to
-include the Rust-powered features like Desktop Access, you'll
-also need `cargo` and `rustc`. The current versions of these tools can be found
-in `build.assets/Makefile`.
+**Important:**
 
-```bash
-# get the source & build:
+* The Go compiler is somewhat sensitive to the amount of memory: you will need
+  **at least** 1GB of virtual memory to compile Teleport. A 512MB instance
+  without swap will **not** work.
+* This will build the latest version of Teleport, **regardless** of whether it
+  is stable. If you want to build the latest stable release, run `git checkout`
+  to the corresponding tag (for example, run `git checkout v8.0.0`) **before**
+  performing a build.
+* The following instructions are Mac-specific.
+
+#### Dependencies
+
+Ensure you have installed correct versions of dependencies:
+
+* `Go` version from [go.mod](https://github.com/gravitational/teleport/blob/0e9d621b464f2bd05a28e7db0ae2ef65349c460e/go.mod#L3)
```

Comment:
Link to master? Picking a fixed commit is likely to yield outdated info over time.

```suggestion
* `Go` version from
  [go.mod](https://github.com/gravitational/teleport/blob/master/go.mod#L3)
```

Reply:
Yeah sorry I'll fix all links to point at master. Got these through "Open in Github" in IDE, which leads to my branch obviously.